Replace blank spaces with underscores in column types. Closes #19. Also update dependencies to more recent versions, and other minor cleanup.

aktary opened the pull request.

aktary committed:
Replace blank spaces with underscores in column types. Closes #19. Also update dependencies to more recent versions, and other minor cleanup.

simonireilly commented on src/adapters/mermaid.ts:
```diff
@@ -66,7 +66,7 @@ export class MermaidErd {
         const columns = entry.columns.map((column) => {
           return [
-            this.dataSource.driver.normalizeType(column),
+            this.dataSource.driver.normalizeType(column).replace(/\s+/g, "_"),
```
@aktary can you give a little explanation for the reason for this change?

It may be very obvious e.g. spaces in column names break the diagraming tool for example.

Just give a bit of detail for the patch notes please.

aktary replied:
Yes, that's the reason. I assumed linking to the GH issue was sufficient. I'll add more detail in this commit.

simonireilly commented on .nvmrc:
```diff
@@ -0,0 +1 @@
+18.10.0
```
Appreciate this change @aktary - I'll need to update the test suite to drop support 🤣

aktary replied:
I've updated the snapshots.

simonireilly commented on tsconfig.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "@tsconfig/node16",
+  "extends": "./node_modules/@tsconfig/node16/tsconfig.json",
```
Can you version this to node18 given the updates to the default node version

aktary replied:
Done.

aktary committed:
The main reason for this PR is to replace whitespaces in column names with underscores because the whitespaces were breaking mermaid.
In this commit:
- Update tsconfig.json to extend @tsconfig/node18;
- remove very old GH CI node.js versions
- update GH CI latest.yml to use node 18.

aktary commented:
Please let me know if there are any other blockers to merging this PR. I'd love to have it merged so I can share your package with the rest of my team.
